### PR TITLE
Application-not-responding detection

### LINKED ISSUE #708: ANR detection of the Unity event loop
The SDK should detect "application-not-responding" based on the Android SDK (need to confirm they are being sent).

Sample can be done by blocking for more than 60 seconds in C#

The Android approach is here: https://github.com/getsentry/sentry-java/blob/main/sentry-android-core/src/main/java/io/sentry/android/core/ANRWatchDog.java

### PR DESCRIPTION
**Pull request by vaind**
* closes #708 
* https://sentry.io/organizations/sentry-sdks/issues/3314229346/events/e83f4535c2294197b393000c2a09452d/?project=5439417

### CODE REVIEW

**Comment by bruno-garcia** [comment]
Added @marandaneto in case he wants to take a look. He's got a lot of experience reducing noise from the Android ANR so this PR could benefit from that

File: src/Sentry.Unity/Integrations/ANRIntegration.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                    _thread = new Thread(Run)
+                    {
+                        Name = "Sentry-ANR-WatchDog"
+                    };
+                    _thread.Start();
+
+                    // Stop the thread when the app is being shut down.
+                    _monoBehaviour.Application.Quitting += () =>
+                    {
+                        _cancel.Cancel();
```
Comment (bruno-garcia) [inline src/Sentry.Unity/Integrations/ANRIntegration.cs:67]: Cancel can throw. Should we have `Join` in a `finally` block and log out on `catch`?

File: src/Sentry.Unity/Integrations/ANRIntegration.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                        }
+
+                        _uiIsResponsive = false;
+                        watch.Restart();
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                _logger?.Log(SentryLevel.Warning, "Exception in the ANR watchdog.", e);
```
Comment (bruno-garcia) [inline src/Sentry.Unity/Integrations/ANRIntegration.cs:123]: Perhaps best to use Level Error?

Comment (vaind) [reply src/Sentry.Unity/Integrations/ANRIntegration.cs:123]: added a catch clause for ThreadAbortException with a warning and changed this to error

File: src/Sentry.Unity/Integrations/ANRIntegration.cs (hunk 1 above)
Comment (vaind) [reply src/Sentry.Unity/Integrations/ANRIntegration.cs]: removed the cancellation token in favour of a plain bool, which in the meantime I've found out is atomic in c#

**Comment by vaind** [comment]
Note: I've refactored the code to make the thread a [background thread](https://docs.microsoft.com/en-us/dotnet/api/system.threading.thread.isbackground?view=netstandard-2.1), to not block during app shutdown

**Review by bruno-garcia — APPROVED** [review]

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**